doc/rules/index.rst

```diff
@@ -919,6 +919,9 @@ Whitespace
 - `no_whitespace_in_blank_line <./whitespace/no_whitespace_in_blank_line.rst>`_
 
   Remove trailing whitespace at the end of blank lines.
+- `reference_spaces <./whitespace/reference_spaces.rst>`_
```

Comment:
before we go into the new fixer, we need to acknowledge that some users will not use new feature, but instead they will use only already existing features, that will lead to confusing style format.

Can I ask you, before we proceed with this new fixer, to first work on bugfix that would prevent existing rules from applying the wrong change?
If yes, please raise that as a separated PR.

I identified those 2 rules that are doing harm:

```diff
   1) x.php (unary_operator_spaces)
 <?php
 
-$a =& $b;
+$a =&$b;
```

```diff
   1) x.php (binary_operator_spaces)
 <?php
 
-$a =& $b;
+$a = & $b;
```